### PR TITLE
HarID

### PR DESCRIPTION
**Pull request by pjotrsavitski**
A [HarID](https://harid.ee/en/pages/docs) provider that allows usage of both **test** and **production** environments by setting a config parameter.

### CODE REVIEW

File: src/HarID/Provider.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    /**
+     * Returns well known configuration for IdP.
+     *
+     * @return array
+     */
+    protected function getWellKnownConfiguration()
+    {
+        $url = $this->useTestIdp() ? 'https://test.harid.ee/.well-known/openid-configuration' : 'https://harid.ee/.well-known/openid-configuration';
+
+        $value = Cache::remember($url, 60 * 60 * 24, function () use ($url) {
```
[the remark this answers is not in the document]
Comment (lucasmichot) [reply src/HarID/Provider.php]: In this case, I would just hardcode them

[the remark this answers is not in the document]
Comment (lucasmichot) [reply src/HarID/Provider.php]: If they ever change, a new version can be introduced

[the remark this answers is not in the document]
Comment (pjotrsavitski) [reply src/HarID/Provider.php]: I'll make the change in about an hour. Then push that version up. Let it be hard-coded for now.

**Comment by pjotrsavitski** [comment]
I think that it should be good enough.

File: src/HarID/Provider.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    /**
+     * Returns base URL for idP endpoints with check for test or live environment.
+     *
+     * @return string
+     */
+    protected function getEndpointBaseUrl()
+    {
+        return ($this->getConfig('use_test_idp', false) === true) ? 'https://test.harid.ee/et' : 'https://harid.ee/et';
```
Comment (lucasmichot) [inline src/HarID/Provider.php:46]:
```suggestion
        return (bool) $this->getConfig('use_test_idp', false) === true ? 'https://test.harid.ee/et' : 'https://harid.ee/et';
```

**Review by lucasmichot — APPROVED** [review]
@pjotrsavitski thanks again for the PR, can you make sure to address my very last comment: https://github.com/SocialiteProviders/Providers/pull/758/files#r739652092

@atymic good to merge 👍

File: src/HarID/Provider.php (hunk 2 above)
Comment (pjotrsavitski) [reply src/HarID/Provider.php:46]: I've had a similar code before and changed that to be strict check as almost any value passed to the `use_test_idp` config setting would make it true. I'm not against the change though as it does not really matter all that much.

**Comment by pjotrsavitski** [comment]
@lucasmichot I've accepted the suggestion as it seems to be more or less the same logic, if someone bothers to define the additional configuration option, then they probably really want the test IdP to be used.

**Review by atymic — APPROVED** [review]

**Pull request merged**